DashboardClient/DashboardClient.hpp

```diff
@@ -92,10 +104,19 @@ namespace Umati {
 			std::shared_ptr<OpcUaTypeReader> m_pTypeReader;
 
 			std::set<ModelOpcUa::NodeId_t> browsedNodes;
+			std::map<const ModelOpcUa::NodeId_t, std::shared_ptr<const ModelOpcUa::SimpleNode>> browsedSimpleNodes;
 			std::recursive_mutex m_dataSetMutex;
 			std::list<std::shared_ptr<DataSetStorage_t>> m_dataSets;
 			std::map<std::string, LastMessage_t> m_latestMessages;
 
+			bool m_eventThreadRunning;
+			std::thread m_eventThread;
+			std::queue<IDashboardDataClient::StructureChangeEvent> m_eventqueue;
```

Comment:
std::queue is not threadsafe! Add mutex/lockfree queue or sth

Reply:
Replaced by a threadsafe queue.